src/kibana/components/index_patterns/_flatten_search_response.js

```diff
@@ -10,7 +10,8 @@ define(function (require) {
         stack.push(key);
         var flattenKey = stack.join('.');
 
-        if ((self.fields.byName[flattenKey] || _.isArray(obj[key]) || !_.isObject(obj[key]))) {
```

Comment:
Yeah.  PR #2723 does exactly that.  This PR should probably wait until after that one, and this one can be refactored on top at that point.